src/tools/split.cpp

```diff
@@ -0,0 +1,103 @@
+#include "split.h"
+#include "editorview.h"
+#include "meshitem.h"
+#include "gripitem.h"
+#include <QMouseEvent>
+#include <QDebug>
+#include <limits>
+
+SplitTool::SplitTool()
+: BaseTool()
+{
+}
+
+void SplitTool::activated(EditorView* editor)
+{
+  if (marker == nullptr) {
+    marker = new MarkerItem();
+    editor->scene()->addItem(marker);
+    marker->setHighlight(QColor("#FF4444"));
+  }
+  GripItem* closestGrip = editor->snapGrip();
+  EdgeItem* closestEdge = editor->snapEdge();
+
+  if (closestEdge || closestGrip) {
+    marker->setVisible(true);
+    if (closestGrip) {
+      marker->setBrush(closestGrip->color());
+    }
+    else {
+      marker->setBrush(closestEdge->colorAt(editor->snapPosition()));
+    }
+    marker->setPos(editor->snapPosition());
+  }
+  else {
+    marker->setVisible(false);
+  }
+}
+
+void SplitTool::deactivated(EditorView* editor)
+{
+    marker->setVisible(false);
+}
+
+bool SplitTool::mousePressEvent(EditorView* editor, QMouseEvent* event)
+{
+  MeshItem* mesh = editor->activeMesh();
+  GripItem* oldActive = editor->activeVertex();
+  GripItem* closestGrip = editor->snapGrip();
+  EdgeItem* closestEdge = editor->snapEdge();
+  if (closestEdge) {
+    if (!closestGrip) {
+      closestEdge->split(editor->snapPosition());
+    }
+    else {
+      editor->setActiveVertex(closestGrip);
+    }
+    if (oldActive) {
+      mesh->splitPolygon(oldActive, editor->activeVertex());
+    }
+  }
+  /* if (grip) { */
+  /*   if (mesh && (event->modifiers() & Qt::ShiftModifier) && mesh->activeVertex() != grip) { */
+  /*     mesh->splitPolygon(mesh->activeVertex(), grip); */
+  /*     return true; */
+  /*   } else { */
+  /*     editor->setActiveVertex(grip); */
+  /*   } */
+  /* } else { */
+  /*   editor->setActiveVertex(nullptr); */
+  /* } */
+  return false;
+}
+
+bool SplitTool::mouseMoveEvent(EditorView* editor, QMouseEvent* event)
+{
+  if (marker == nullptr) {
```

Comment:
Wrap this in a function instead of duplicating code

Reply:
That was on the todo list. Wasn't sure on function name for it or if it should be split into multiple functions or what.

Reply:
I'd probably go with `MarkerItem* SplitTool::marker()` and rename the variable.

Reply:
As in a public marker() function?

Reply:
Yeah. Can't be `const` in this case though!